mangaki/mangaki/fields.py

```diff
@@ -0,0 +1,56 @@
+from django.db import models
+
+from mangaki.choices import Category
+
+# See Django custom fields documentation at
+# https://docs.djangoproject.com/en/1.9/howto/custom-model-fields/j
+class CategoryField(models.IntegerField):
+    """
+    A custom field that handles automatically conversion between an integer
+    (stored in the database) and a convenient Category object.
+
+    This field handles conversion from integer, strings, and Category objects,
+    allowing the following kind of patterns:
+
+        Work.objects.filter(category=1)
+        Work.objects.filter(category='anime')
+        Work.objects.last().category.slug # 'anime', 'manga', etc.
+        str(Work.objects.last().category) # 'Anime', 'Manga', etc.
+    """
+
+    description = "A work category"
+
+    def __init__(self, *args, **kwargs):
+        kwargs['choices'] = Category.CHOICES.items()
+        super().__init__(*args, **kwargs)
+
+    def get_prep_value(self, value):
+        if isinstance(value, str):
+            value = Category(value)
+
+        return super().get_prep_value(value)
+
+    def get_prep_lookup(self, lookup_type, value):
+        if lookup_type == 'exact':
+            return self.get_prep_value(value)
+        elif lookup_type == 'in':
+            return [self.get_prep_value(v) for v in value]
+        else:
+            raise TypeError('Lookup type %r not supported.' % lookup_type)
```

Comment:
Comme ça on a des guillemets for free. D'une façon générale, dans les messages d'erreur, `%r` c'est mieux, parcequ'on préfère être précis sur ce dont on parle plutôt que de laisser le programmeur perplexe quand à la signification d'un `__str__` destiné à un pretty-printing pas suffisement précis.